backend/sirius-web-spring-collaborative-diagrams/src/main/java/org/eclipse/sirius/web/spring/collaborative/diagrams/handlers/InvokeNodeToolOnDiagramEventHandler.java

```diff
@@ -104,9 +106,9 @@ public void handle(One<IPayload> payloadSink, Many<ChangeDescription> changeDesc
                     .map(CreateNodeTool.class::cast);
             // @formatter:on
             if (optionalTool.isPresent()) {
-                Status status = this.executeTool(editingContext, diagramContext, input.getDiagramElementId(), optionalTool.get(), input.getStartingPositionX(), input.getStartingPositionY(),
+                IStatus status = this.executeTool(editingContext, diagramContext, input.getDiagramElementId(), optionalTool.get(), input.getStartingPositionX(), input.getStartingPositionY(),
                         input.getSelectedObjectId());
-                if (Objects.equals(status, Status.OK)) {
+                if (Objects.equals(status, new Success())) {
```

Comment:
Use `instanceof` again, I'm not sure why you have difficulties properly testing the type of an object